### PR TITLE
Always guarantee realname is removed

### PR DESCRIPTION
**Pull request by Universal-Omega**
May be some circumstances it isn't, this should guarantee that it is.

### CODE REVIEW

**Comment by Reception123** [comment]
@Universal-Omega Hmm, seems checks are failing

**Comment by Universal-Omega** [comment]
> @Universal-Omega Hmm, seems checks are failing

Yeah, unrelated though. I know why. Its an upstream issue with latest docker image tag.

**Comment by dmehus** [comment]
@Universal-Omega, thanks! Would it be helpful to add @JohnFLewis as a reviewer on this task to identify the issue?

**Comment by Universal-Omega** [comment]
> @Universal-Omega, thanks! Would it be helpful to add @JohnFLewis as a reviewer on this task to identify the issue?

I know the issue. That's why I was able to do this PR. If I didn't know, I wouldn't be able to do it.

**Comment by dmehus** [comment]
> > @Universal-Omega, thanks! Would it be helpful to add @JohnFLewis as a reviewer on this task to identify the issue?
> 
> I know the issue. That's why I was able to do this PR. If I didn't know, I wouldn't be able to do it.

Okay, thanks. Sounds good. :)

File: includes/RemovePIIJob.php [hunk 1]
```diff
@@ -523,8 +523,15 @@ public function run() {
 		// Remove user email
 		$userLatest = $newName->getInstanceForUpdate();
 
-		if ( $userLatest->getEmail() ) {
-			$userLatest->invalidateEmail();
+		if ( $userLatest->getEmail() || $userLatest->getRealName() ) {
```
Comment (JohnFLewis) [inline includes/RemovePIIJob.php:526]: Running the same command twice seems wasteful, can't this either be stored/cached or reduce to one call?

Comment (Universal-Omega) [reply includes/RemovePIIJob.php]: Thanks! Done. Took a look and it seems saveSettings can even be called if no settings are changed, so I just left thar, removing the initial check, and always saveSettings.

**Pull request merged**